submissions/api.py

```diff
@@ -865,6 +865,58 @@ def set_score(submission_uuid, points_earned, points_possible,
         pass
 
 
+def get_download_status(student_item_dict, staff_user):
+    """
+    Check if the submission was downloaded by a staff user and return download status.
+
+    Args:
+        student_item_dict (dict): StudentItem data.
+        staff_user (User): The staff user to check the submission download status for.
+
+    Returns:
+        True if the submission was downloaded, False if not
+    """
+    student_item = _get_or_create_student_item(student_item_dict)
+    try:
+        submission = Submission.objects.filter(student_item=student_item).last()
+        if submission.staff_downloads:
+            staff_users = submission.staff_downloads.get('users', [])
+            if staff_user.username in staff_users:
+                return True
+    except DatabaseError as e:
+        logger.exception('{}: {}'.format(type(e), e.message))
+    return False
```

Comment:
This will fail silently if the submission is not saved (which shouldn't happen). Maybe it's better to reraise the exception with a custom message?

Reply:
@iivic you missed the point of this code. First of all, no saving is being done here.
However, I do think the exception raising is unnecessary. What's the exception (case) you're trying to prevent here?